tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -216,6 +219,21 @@ def test_info(self):
     self.assertEqual(self.builder.name, info.name)
 
   def _get_dl_extract_result(self, url):
+
+    def add_url(url_or_urls):
+      # url_or_urls: url or `list`/`dict` of urls to download and extract. 
+      # Each url can be a `str` or `tfds.download.Resource`.
+      if isinstance(url_or_urls, dict):
+        for _, v in url_or_urls.items():
+          add_url(v)
+      elif isinstance(url_or_urls, list):
+        map(add_url, url_or_urls)
+      elif isinstance(url_or_urls, download.resource.Resource):
+        self._download_urls.add(url_or_urls.url)
+      else:
+        self._download_urls.add(url_or_urls)
+
+    add_url(url)
```

Comment:
You could use `tf.nest.map_structure` instead to recursively add the urls.

And you also need to modify `_get_dl_download_result `